packages/hint/src/lib/utils/dom/new-traverse.ts

```diff
@@ -0,0 +1,38 @@
+import { NewAsyncHTMLDocument, NewAsyncHTMLElement } from '../../types/new-async-html';
```

Comment:
change the filename to `traverse` and similar with `new-async-html`?

Reply:
working on it. I'm calling them `shadow-traverse`, `ShadowAsyncHTMLDocument`, etc.

Reply:
Not sure about the `shadow` part as it could be confused with `shadow-dom`. I might just leave it as `traverse`, `HTMLDocument` and such. @antross what do you think?

Reply:
I agree that `shadow` is too likely to be confused with `shadow-dom`. Let's stick with `traverse` for now unless someone has another idea.